// ngx-simple-modal/dist/simple-modal/simple-modal.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var rxjs_1 = require("rxjs");
var SimpleModalComponent = (function () {
    function SimpleModalComponent() {
    }
    SimpleModalComponent.prototype.mapDataObject = function (data) {
        data = data || {};
        var keys = Object.keys(data);
        for (var i = 0, length_1 = keys.length; i < length_1; i++) {
            var key = keys[i];
            this[key] = data[key];
        }
    };
    SimpleModalComponent.prototype.setupObserver = function () {
        var _this = this;
        return rxjs_1.Observable.create(function (observer) {
            _this.observer = observer;
            _this.completeOnDestroy(_this);
            return function () {
                _this.close();
            };
        });
    };
    SimpleModalComponent.prototype.onClosing = function (callback) {
        this.closerCallback = callback;
    };
    SimpleModalComponent.prototype.close = function () {
        return this.closerCallback(this);
    };
    SimpleModalComponent.prototype.onKeydownHandler = function (evt) {
        if (this.options && this.options.closeOnEscape) {
            this.close();
        }
    };
    SimpleModalComponent.prototype.completeOnDestroy = function (component) {
        var _this = this;
        var ngDestroyOriginal = component.ngOnDestroy;
        component.ngOnDestroy = function () {
            if (ngDestroyOriginal) {
                ngDestroyOriginal.apply(component);
            }
            if (_this.observer) {
                _this.observer.next(_this.result);
                _this.observer.complete();
            }
        };
    };
    SimpleModalComponent.prototype.ngOnDestroy = function () {
    };
    SimpleModalComponent.propDecorators = {
        onKeydownHandler: [{ type: core_1.HostListener, args: ['document:keydown.escape', ['$event'],] }]
    };
    return SimpleModalComponent;
}());
exports.SimpleModalComponent = SimpleModalComponent;
