// ngx-simple-modal/dist/simple-modal/simple-modal-holder.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var simple_modal_options_1 = require("./simple-modal-options");
var simple_modal_wrapper_component_1 = require("./simple-modal-wrapper.component");
var SimpleModalHolderComponent = (function () {
    function SimpleModalHolderComponent(resolver, defaultSimpleModalOptions) {
        this.resolver = resolver;
        this.defaultSimpleModalOptions = defaultSimpleModalOptions;
        this.modals = [];
    }
    SimpleModalHolderComponent.prototype.addModal = function (component, data, options) {
        var _this = this;
        var factory = this.resolver.resolveComponentFactory(simple_modal_wrapper_component_1.SimpleModalWrapperComponent);
        var componentRef = this.viewContainer.createComponent(factory);
        var modalWrapper = componentRef.instance;
        var _component = modalWrapper.addComponent(component);
        _component.options = options = Object.assign({}, this.defaultSimpleModalOptions, options);
        modalWrapper.modalClasses = options.wrapperDefaultClasses;
        this.modals.push(_component);
        this.wait().then(function () {
            _this.toggleWrapperClass(modalWrapper.wrapper, options.wrapperClass);
            _this.toggleBodyClass(options.bodyClass);
        });
        _component.onClosing(function (modal) { return _this.removeModal(modal); });
        this.configureCloseOnClickOutside(modalWrapper);
        _component.mapDataObject(data);
        return _component.setupObserver();
    };
    SimpleModalHolderComponent.prototype.removeModal = function (closingModal) {
        var _this = this;
        var options = closingModal.options;
        this.toggleWrapperClass(closingModal.wrapper, options.wrapperClass);
        return this.wait(options.animationDuration).then(function () {
            _this.removeModalFromArray(closingModal);
            _this.toggleBodyClass(options.bodyClass);
        });
    };
    SimpleModalHolderComponent.prototype.removeAllModals = function () {
        var _this = this;
        return Promise.all(this.modals.map(function (modal) { return _this.removeModal(modal); }));
    };
    SimpleModalHolderComponent.prototype.toggleBodyClass = function (bodyClass) {
        if (!bodyClass) {
            return;
        }
        var body = document.getElementsByTagName('body')[0];
        var bodyClassItems = bodyClass.split(' ');
        if (!this.modals.length) {
            (_a = body.classList).remove.apply(_a, bodyClassItems);
        }
        else {
            (_b = body.classList).add.apply(_b, bodyClassItems);
        }
        var _a, _b;
    };
    SimpleModalHolderComponent.prototype.configureCloseOnClickOutside = function (modalWrapper) {
        if (modalWrapper.content.options.closeOnClickOutside) {
            modalWrapper.onClickOutsideModalContent(modalWrapper.content.options.closeOnClickOutside, function () {
                modalWrapper.content.close();
            });
        }
    };
    SimpleModalHolderComponent.prototype.toggleWrapperClass = function (modalWrapperEl, wrapperClass) {
        var wrapperClassList = modalWrapperEl.nativeElement.classList;
        var wrapperClassItems = wrapperClass.split(' ');
        if (wrapperClassList.toString().indexOf(wrapperClass) !== -1) {
            wrapperClassList.remove.apply(wrapperClassList, wrapperClassItems);
        }
        else {
            wrapperClassList.add.apply(wrapperClassList, wrapperClassItems);
        }
    };
    SimpleModalHolderComponent.prototype.wait = function (ms) {
        if (ms === void 0) { ms = 0; }
        return new Promise(function (resolve, reject) {
            setTimeout(function () { return resolve(); }, ms);
        });
    };
    SimpleModalHolderComponent.prototype.removeModalFromArray = function (component) {
        var index = this.modals.indexOf(component);
        if (index > -1) {
            this.viewContainer.remove(index);
            this.modals.splice(index, 1);
        }
    };
    SimpleModalHolderComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'simple-modal-holder',
                    template: '<ng-template #viewContainer></ng-template>',
                },] },
    ];
    SimpleModalHolderComponent.ctorParameters = function () { return [
        { type: core_1.ComponentFactoryResolver },
        { type: simple_modal_options_1.SimpleModalOptions, decorators: [{ type: core_1.Inject, args: [simple_modal_options_1.DefaultSimpleModalOptionConfig,] }] }
    ]; };
    SimpleModalHolderComponent.propDecorators = {
        viewContainer: [{ type: core_1.ViewChild, args: ['viewContainer', { read: core_1.ViewContainerRef },] }]
    };
    return SimpleModalHolderComponent;
}());
exports.SimpleModalHolderComponent = SimpleModalHolderComponent;
